### ISSUE TITLE
Updating pipeline schedules is not working

### ISSUE THREAD

**Issue by stuart-warren** [gdubicki/gitlabform#364]
**Describe the bug**

Similar to #361 updating pipeline schedules seems to have the same issue

```py
Processing section 'schedules'
Error: Error occurred while processing project foo/gitlab/gitlabform, exception:
Request url='https://gitlab.foo.tech//api/v4/projects/foo%2Fgitlab%2Fgitlabform/pipeline_schedules/2464', method=PUT, data='{"active": true, "cron": "0 10 * * MON-FRI", "cron_timezone": "London", "ref": "master"}' failed - expected code(s) [200, 201], got code 403 & body: '{"message":"403 Forbidden"}'
Traceback (most recent call last):
  File "/gitlabform/gitlabform/core.py", line 430, in run
    self.project_processors.process_entity(
  File "/gitlabform/gitlabform/processors/__init__.py", line 30, in process_entity
    processor.process(
  File "/gitlabform/gitlabform/processors/util/decorators.py", line 42, in method_wrapper
    return method(self, project_and_group, SafeDict(configuration), *args)
  File "/gitlabform/gitlabform/processors/abstract_processor.py", line 48, in process
    self._process_configuration(project_or_project_and_group, configuration)
  File "/gitlabform/gitlabform/processors/project/schedules_processor.py", line 38, in _process_configuration
    updated_schedule = self.gitlab.update_pipeline_schedule(
  File "/gitlabform/gitlabform/gitlab/schedules.py", line 39, in update_pipeline_schedule
    return self._make_requests_to_api(
  File "/gitlabform/gitlabform/gitlab/core.py", line 127, in _make_requests_to_api
    response = self._make_request_to_api(
  File "/gitlabform/gitlabform/gitlab/core.py", line 218, in _make_request_to_api
    raise UnexpectedResponseException(
gitlabform.gitlab.core.UnexpectedResponseException: Request url='https://gitlab.foo.tech//api/v4/projects/foo%2Fgitlab%2Fgitlabform/pipeline_schedules/2464', method=PUT, data='{"active": true, "cron": "0 10 * * MON-FRI", "cron_timezone": "London", "ref": "master"}' failed - expected code(s) [200, 201], got code 403 & body: '{"message":"403 Forbidden"}'
```

**GitLabForm version**

`2.11.0`

**GitLab version**

F.e. `14.9.4-ee`

**Comment by stuart-warren** [comment]
Ah, think this may be due to an upstream change needing more than maintainer access to update pipeline schedules

https://gitlab.com/gitlab-org/gitlab-foss/blob/master/CHANGELOG.md#14101-2022-04-29

**Comment by stuart-warren** [comment]
This was due to the gitlabform bot not being the owner of the scheduled pipeline it was trying to update

Curious it doesn't set the owner before trying to update the schedule

**Comment by stuart-warren** [comment]
Perhaps these need to be swapped

https://github.com/gdubicki/gitlabform/blob/3adb51f673e910a7ed93ce7aa81d0454778dda19/gitlabform/processors/project/schedules_processor.py#L43

**Comment by gdubicki** [comment]
Hi @stuart-warren!

I wonder why this hasn't been caught by our acceptance tests... Anyway, I did what you suggested. Can you please test [v2.11.1rc1](https://github.com/gdubicki/gitlabform/releases/tag/v2.11.1rc1) pre-release with the fix?

**Comment by stuart-warren** [comment]
You probably don't have @jocado manually making themselves an owner of your scheduled pipelines :D

**Comment by stuart-warren** [comment]
Unfortunately, this fails. Have added comment to MR

**Comment by stuart-warren** [comment]
rc2 seems to work for me, thanks

**Comment by gdubicki** [comment]
I just released v2.11.1 final with the fix.
